lib/Backend/CMakeLists.txt

```diff
@@ -1,44 +1,44 @@
-set(CURRENT_HEADER_DIR "${CMAKE_CURRENT_SOURCE_DIR}/../../include/hipacc/Backend")
-
-set(Backend_Generators_SOURCES CPU_x86.cpp CUDA.cpp OpenCL_ACC.cpp OpenCL_CPU.cpp OpenCL_GPU.cpp)
-set(Backend_Generators_HEADERS "${CURRENT_HEADER_DIR}/CPU_x86.h"
-                               "${CURRENT_HEADER_DIR}/CUDA.h"
-                               "${CURRENT_HEADER_DIR}/OpenCL_ACC.h"
-                               "${CURRENT_HEADER_DIR}/OpenCL_CPU.h"
-                               "${CURRENT_HEADER_DIR}/OpenCL_GPU.h")
-
-set(Backend_Generators_Base_HEADERS "${CURRENT_HEADER_DIR}/AcceleratorDeviceBase.h"
-                                    "${CURRENT_HEADER_DIR}/CodeGeneratorBaseImplT.h")
-
-set(Backend_Common_SOURCES ClangASTHelper.cpp InstructionSets.cpp)
-
-set(Backend_Common_HEADERS "${CURRENT_HEADER_DIR}/BackendExceptions.h"
-                           "${CURRENT_HEADER_DIR}/ClangASTHelper.h"
-                           "${CURRENT_HEADER_DIR}/CommonDefines.h"
-                           "${CURRENT_HEADER_DIR}/ICodeGenerator.h"
-                           "${CURRENT_HEADER_DIR}/InstructionSets.h"
-                           "${CURRENT_HEADER_DIR}/OptionParsers.h"
-                           "${CURRENT_HEADER_DIR}/VectorizationAST.h"
-                           "${CURRENT_HEADER_DIR}/Vectorizer.h")
-
-set(Backend_SOURCES BackendConfigurationManager.cpp VectorizationAST.cpp Vectorizer.cpp ${Backend_Common_SOURCES} ${Backend_Generators_SOURCES})
-set(Backend_HEADERS "${CURRENT_HEADER_DIR}/BackendConfigurationManager.h"
-                    ${Backend_Common_HEADERS}
-                    ${Backend_Generators_HEADERS}
-                    ${Backend_Generators_Base_HEADERS})
-
-add_library(hipaccBackend ${Backend_SOURCES} ${Backend_HEADERS})
-
-source_group("Source Files\\Common" FILES ${Backend_Common_SOURCES})
-source_group("Source Files\\Generators" FILES ${Backend_Generators_SOURCES})
-source_group("Header Files\\Common" FILES ${Backend_Common_HEADERS})
-source_group("Header Files\\Generators" FILES ${Backend_Generators_HEADERS})
-source_group("Header Files\\Generators\\Base" FILES ${Backend_Generators_Base_HEADERS})
+# ./lib/Backend/CMakeLists.txt
+
+add_library(hipaccBackend BackendConfigurationManager.cpp)
+add_cppcheck(hipaccBackend POSSIBLE_ERROR FAIL_ON_WARNINGS)
+target_sources(hipaccBackend
+  PRIVATE VectorizationAST.cpp
+          Vectorizer.cpp
+          CPU_x86.cpp
+          CUDA.cpp
+          OpenCL_ACC.cpp
+          OpenCL_CPU.cpp
+          OpenCL_GPU.cpp
+          ClangASTHelper.cpp
+          InstructionSets.cpp)
+target_include_directories(hipaccBackend SYSTEM
```

Comment:
The Hipacc includes are not system include paths. It guess we should remove the `SYSTEM` keyword here?

Reply:
${LLVM_INCLUDE_DIRS) is?

Reply:
Yes and no ... basically, the rationale behind using `SYSTEM` was to suppress warnings from LLVM includes.
`SYSTEM` is ignored on Windows and might cause problems on Linux depending on the LLVM installation directory.

Reply:
@oreiche @richardmembarth, do you suggest to use SYSTEM or not? I removed it and warnings from clang and llvm includes explode...

Reply:
To be honest, I do not really understand the problem without knowing the exact warnings that are reported. If `SYSTEM` is the only way to get rid of them, then add it. However, adding them only for `${LLVM_INCLUDE_DIRS}` should be sufficient right?

Reply:
The compiler is emitting warnings for the LLVM headers, not our code.
For this reason it makes sense to use `SYSTEM` for now.

Reply:
yes, warnings are from LLVM and Clang headers, some examples:
```
/scratch-local/usr/local/include/llvm/Support/TrailingObjects.h: In instantiation of ‘static size_t llvm::TrailingObjects<BaseTy, TrailingTys>::callNum        TrailingObjects(const BaseTy*, llvm::trailing_objects_internal::TrailingObjectsBase::OverloadToken<BaseTy>) [with BaseTy = clang::ConstantArrayType; Tr        ailingTys = {const clang::Expr*}; size_t = long unsigned int]’:
/scratch-local/usr/local/include/llvm/Support/TrailingObjects.h:181:55:   required from ‘static NextTy* llvm::trailing_objects_internal::TrailingObject        sImpl<Align, BaseTy, TopTrailingObj, PrevTy, NextTy, MoreTys ...>::getTrailingObjectsImpl(BaseTy*, llvm::trailing_objects_internal::TrailingObjectsBase        ::OverloadToken<NextTy>) [with int Align = 8; BaseTy = clang::ConstantArrayType; TopTrailingObj = llvm::TrailingObjects<clang::ConstantArrayType, const         clang::Expr*>; PrevTy = clang::ConstantArrayType; NextTy = const clang::Expr*; MoreTys = {}]’ 

/scratch-local/usr/local/include/clang/AST/Expr.h: In constructor ‘clang::ImplicitCastExpr::ImplicitCastExpr(clang::ImplicitCastExpr::OnStack_t, clang:        :QualType, clang::CastKind, clang::Expr*, clang::ExprValueKind)’:
/scratch-local/usr/local/include/clang/AST/Expr.h:3288:30: warning: unused parameter ‘_’ [-Wunused-parameter] ImplicitCastExpr(OnStack_t _, QualType ty, CastKind kind, Expr *op
/scratch-local/usr/local/include/clang/AST/Expr.h: In constructor ‘clang::BinaryOperator::BinaryOperator(clang::Expr*, clang::Expr*, clang::BinaryOpera        tor::Opcode, clang::QualType, clang::ExprValueKind, clang::ExprObjectKind, clang::SourceLocation, clang::FPOptions, bool)’:
``` 
then i will use `SYSTEM` for now, only for `${LLVM_INCLUDE_DIRS)`.

Reply:
Oh boy, what are they doing...? Doesn't clang use clang-tidy?? ^^